pkgs/build-support/vm/default.nix

```diff
@@ -985,25 +1014,287 @@ rec {
+          hash = "sha256-MskwZBQtifPxnBHpJkLFq9g2hBj3qz873XUuSvqbWyM=";
+        })
+      ];
+      urlPrefixes = [
+        "https://dl.rockylinux.org/vault/rocky/10.0/BaseOS/x86_64/os"
+        "https://dl.rockylinux.org/vault/rocky/10.0/AppStream/x86_64/os"
+      ];
+      archs = [
+        "noarch"
+        "x86_64"
+      ];
+      packages = commonRockyPackages ++ [
+        "annobin-plugin-gcc"
+      ];
+      unifiedSystemDir = true;
+    };
+
+    alma9x86_64 = {
+      name = "alma-9-x86_64";
+      fullName = "AlmaLinux 9 (x86_64)";
+      packagesLists = [
+        (fetchurl {
+          url = "https://repo.almalinux.org/almalinux/9/BaseOS/x86_64/os/repodata/e7e7e537062ffd6778b5190b7b6785086efab6647401014d1985c10a0d3de609-primary.xml.gz";
+          hash = "sha256-5+flNwYv/Wd4tRkLe2eFCG76tmR0AQFNGYXBCg095gk=";
+        })
+        (fetchurl {
+          url = "https://repo.almalinux.org/almalinux/9/AppStream/x86_64/os/repodata/e61f480b84cb0a671452c21cbd52930a6fa111e7fb3407ed75cabff71c1a07b7-primary.xml.gz";
+          hash = "sha256-5h9IC4TLCmcUUsIcvVKTCm+hEef7NAftdcq/9xwaB7c=";
+        })
+      ];
+      urlPrefixes = [
+        "https://repo.almalinux.org/almalinux/9/BaseOS/x86_64/os"
+        "https://repo.almalinux.org/almalinux/9/AppStream/x86_64/os"
+      ];
+      archs = [
+        "noarch"
+        "x86_64"
+      ];
+      packages = commonAlmaPackages ++ [
+        "annobin"
+      ];
+      unifiedSystemDir = true;
+    };
+
+    alma10x86_64 = {
+      name = "alma-10-x86_64";
+      fullName = "AlmaLinux 10 (x86_64)";
+      packagesLists = [
+        (fetchurl {
+          url = "https://repo.almalinux.org/almalinux/10/BaseOS/x86_64/os/repodata/e0a4a18c4f302fa3188b757197b81d606199eccddd8480602c378def572fabbf-primary.xml.gz";
+          hash = "sha256-4KShjE8wL6MYi3Vxl7gdYGGZ7M3dhIBgLDeN71cvq78=";
+        })
+        (fetchurl {
+          url = "https://repo.almalinux.org/almalinux/10/AppStream/x86_64/os/repodata/fb3e7acc93f1b0969c3c00dc3bc4364d4f5837d11274cae650c34cf71c0f80ae-primary.xml.gz";
+          hash = "sha256-+z56zJPxsJacPADcO8Q2TU9YN9ESdMrmUMNM9xwPgK4=";
+        })
+      ];
+      urlPrefixes = [
+        "https://repo.almalinux.org/almalinux/10/BaseOS/x86_64/os"
+        "https://repo.almalinux.org/almalinux/10/AppStream/x86_64/os"
+      ];
+      archs = [
+        "noarch"
+        "x86_64"
+      ];
+      packages = commonAlmaPackages ++ [
+        "annobin-plugin-gcc"
+      ];
+      unifiedSystemDir = true;
+    };
+
+    oracle9x86_64 = {
+      name = "oracle-9-x86_64";
+      fullName = "Oracle Linux 9 (x86_64)";
+      packagesLists = [
+        (fetchurl {
+          url = "https://yum.oracle.com/repo/OracleLinux/OL9/baseos/latest/x86_64/repodata/edda5aa6dc1ad233cd5fe29838e535ef959cd291be343844e127e29c290f93ca-primary.xml.gz";
```

Comment:
also short-lived. i believe you can use the dot-releases though, e.g. <https://yum.oracle.com/repo/OracleLinux/OL9/7/baseos/base/x86_64/repodata/>. Oracle seems to do a good job keeping those around, e.g. [6.1 from 2013](https://yum.oracle.com/repo/OracleLinux/OL6/1/base/x86_64/repodata/) is still up.